autoscaler/controllers/datacollection/daemonset.go

```diff
@@ -289,3 +294,35 @@ func patchDaemonSet(existing *appsv1.DaemonSet, desired *appsv1.DaemonSet, ctx c
 func boolPtr(b bool) *bool {
 	return &b
 }
+
+type DelayManager struct {
+	mu         sync.Mutex
+	inProgress bool
+}
+
+// runFunctionWithDelayAndSkipNewCalls runs the function with the specified delay and skips new calls until the function execution is finished
+func (dm *DelayManager) runFunctionWithDelayAndSkipNewCalls(delay time.Duration, fn func(args ...interface{}) (*appsv1.DaemonSet, error), fnArgs ...interface{}) {
```

Comment:
I'd prefer to use a library function for this if possible.
Looks like https://pkg.go.dev/k8s.io/apimachinery@v0.30.2/pkg/util/wait#PollUntilContextTimeout might be relevant or other functions from that lib